reactive-form-demo/src/app/registration/registration.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, OnInit } from '@angular/core';
+
+import { FormArray, FormControl, FormGroup, FormBuilder, Validators, AbstractControl } from '@angular/forms';
+
+
+@Component({
+  selector: 'app-registration',
+  templateUrl: './registration.component.html',
+  styleUrls: ['./registration.component.css']
+})
+export class RegistrationComponent implements OnInit {
+
+  profileForm = new FormGroup({
+    firstName: new FormControl(''),
+    email: new FormControl(''),
+    password: new FormControl(''),
+    mobileNo: new FormControl('')
+  });
+
+  change: any;
+  staticJsonData: any;
+  newControllers: any;
+  show: boolean;
+
+  constructor(private formBuilder: FormBuilder) {
+    this.profileForm = this.formBuilder.group({
+      firstName: ['',
+        [
+          Validators.required,
+          Validators.minLength(3)
+        ]
+      ],
+      email: ['',
+        [
+          Validators.required,
+          Validators.email
+        ]
+      ],
+      password: ['',
+        [
+          Validators.required,
+          Validators.minLength(8),
+          Validators.pattern(/^(?=.*[A-Z])(?=.*[0-9])(?=.*[@$!%*?&])[A-Za-z0-9@$!%*?&]{8,}$/)
```

Comment:
Create and use custom validator over here

Reply:
custom validator added in new commit